CompoundV3-Bots/Transfers-Monitor/src/agent.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  Finding,
+  Initialize,
+  HandleTransaction,
+  TransactionEvent,
+  ethers,
+} from "forta-agent";
+import { NetworkManager } from "forta-agent-tools";
+import { NetworkData, TransferLog } from "./utils";
+import { BUY_COLLATERAL_ABI, SUPPLY_ABI, TRANSFER_ABI } from "./constants";
+import CONFIG from "./agent.config";
+import { BigNumber, getDefaultProvider } from "ethers";
+import { createTransferFinding } from "./finding";
+
+const networkManager = new NetworkManager(CONFIG);
+
+export const provideInitialize = (
+  networkManager: NetworkManager<NetworkData>,
+  provider: ethers.providers.Provider
+): Initialize => {
+  return async () => {
+    await networkManager.init(provider);
+  };
+};
+
+export const provideHandleTransaction = (
+  networkManager: NetworkManager<NetworkData>
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const transferEvents: Set<any> = new Set([]);
+
+    // Filter the transaction logs for Transfer events in base tokens.
+    txEvent
+      .filterLog(TRANSFER_ABI, networkManager.get("baseTokens"))
+      .forEach((log) => {
+        let cometAddressIndex = networkManager
+          .get("cometAddresses")
+          .indexOf(log.args.to);
+
+        // If the transfer event is destined for a Comet contract with the same baseToken,
+        // add it to the transferEvents set
+        if (
+          cometAddressIndex > -1 &&
+          networkManager.get("cometAddresses")[cometAddressIndex] ==
+            log.args.to &&
+          log.address == networkManager.get("baseTokens")[cometAddressIndex]
+        ) {
+          transferEvents.add({
+            from: log.args.from,
+            index: cometAddressIndex,
+            amount: log.args.amount,
+          });
+        }
```

Comment:
If you do the config change and use an array instead of a set, it could be something like this:

```suggestion
        const cometContracts = networkManager.get("cometContracts");
        const destinationComet = cometContracts.find(comet => comet.address.toLowerCase() === log.args.to.toLowerCase());
        
        if (destinationComet && destinationComet.baseToken.toLowerCase() === log.address.toLowerCase()) {
          transferEvents.push({
            destinationComet,
            from: log.args.from,
            amount: log.args.amount,
          });
        }
```